### PR TITLE
#221 Add generic methods to RichConfig

### PR DESCRIPTION
**Pull request by dmartyanov**
util/ConfigUtil and unicomplex/ConfigUtil enhancements
- added method `def getOption[T](path)` to `RichConfig` class
- added method `def get[T](path,  default)` to `RichConfig` class
- added method `def get[T](path)` to `RichConfig` class
- added tests for `Regex` config value

### CODE REVIEW

**Comment by anilgursel** [comment]
LGTM

**Comment by akara** [comment]
The code looks all good to me. Only one issue. This PR should reflect #221 and not #217. Please change your PR name accordingly. Other than that I'm good to merge. Thanks a ton!

File: squbs-unicomplex/src/test/scala/org/squbs/unicomplex/ConfigUtilSpec.scala [hunk 1]
```diff
@@ -59,69 +62,183 @@ class ConfigUtilSpec extends FunSpecLike with Matchers {
...
+      )
+    }
+
+    it ("should throw WrongType exception for incorrect type of existing config value") {
+      intercept[ConfigException.WrongType](
+        config.get[Int]("testConfig.str")
+      )
+    }
+
+    it ("should get provide at least one IPv$ address for any host") {
```
Comment (akara) [inline squbs-unicomplex/src/test/scala/org/squbs/unicomplex/ConfigUtilSpec.scala]: Is this a typo?

File: squbs-pattern/src/test/scala/org/squbs/pattern/util/ConfigUtilSpec.scala [hunk 2]
```diff
@@ -59,66 +59,156 @@ class ConfigUtilSpec extends FunSpecLike with Matchers {
...
+      )
+    }
+
+    it ("should throw WrongType exception for incorrect type of existing config value") {
+      intercept[ConfigException.WrongType](
+        config.get[Int]("testConfig.str")
+      )
+    }
+
     it ("should get provide at least one IPv$ address for any host") {
```
Comment (akara) [inline squbs-pattern/src/test/scala/org/squbs/pattern/util/ConfigUtilSpec.scala]: Ah, looks like a copy of the typo.

**Comment by akara** [comment]
Thank you very much @dmartyanov!

**Pull request merged**